ipfs-cluster-service/main.go

```diff
@@ -231,6 +242,74 @@ configuration.
 			Usage:  "run the IPFS Cluster peer (default)",
 			Action: daemon,
 		},
+		{
+			Name:   "migration",
+			Usage:  "migrate the IPFS Cluster state between versions",
+			Description: `
+
+This command migrates the internal state of the ipfs-cluster node to the state 
+specified in a backup file. The state format is migrated from the version 
+of the backup file to the version supported by the current cluster version. 
+To succesfully run a migration of an entire cluster, shut down each peer without
+removal, migrate using this command, and restart every peer.
+`,
+			ArgsUsage: "<BACKUP-FILE-PATH> [RAFT-DATA-DIR]",
+			Action: func(c *cli.Context) error {
+				if c.NArg() < 1 || c.NArg() > 2 {
+					return fmt.Errorf("Usage: <BACKUP-FILE-PATH> [RAFT-DATA-DIR]")
+					
+				}
+				//Load configs
+				cfg, clusterCfg, _, _, consensusCfg, _, _, _ := makeConfigs()
+				err := cfg.LoadJSONFromFile(configPath)
+				checkErr("loading configuration", err)
+				backupFilePath := c.Args().First()
+				var raftDataPath string
+				if c.NArg() == 1 {
+					raftDataPath = consensusCfg.DataFolder
+				} else {
+					raftDataPath = c.Args().Get(1)
+				}
+				
+				//Migrate backup to new state
+				backup, err := os.Open(backupFilePath)
+				checkErr("opening backup file", err)
+				defer backup.Close()
+				r := bufio.NewReader(backup)
+				newState := mapstate.NewMapState()
+				err = newState.Restore(r)
+				checkErr("migrating state to alternate version", err)
+				// Encode state to save as a snapshot
+				newStateBytes, err := encodeState(*newState)
+				checkErr("encoding new state", err)
+						
+				//Remove raft state
+				err = cleanupRaft(raftDataPath)
+				checkErr("cleaningup Raft", err)
+				
+				//Write snapshot of the migrated state
+				snapshotStore, err := hraft.NewFileSnapshotStoreWithLogger(raftDataPath, 5, nil)
+				checkErr("creating snapshot store", err)
+				dummyHost, err := makeDummyHost(clusterCfg)
+				checkErr("creating dummy host for snapshot store", err)
+				dummyTransport, err := libp2praft.NewLibp2pTransport(dummyHost, consensusCfg.NetworkTimeout)
```

Comment:
I think you can use `hraft.InmemTransport` rather than going through the trouble of creating a libp2p one?